Add rules C409 and C410 that complain about unnecessary list/tuple

jdufresne opened the pull request.

pope1ni commented on flake8_comprehensions.py:
```diff
@@ -27,6 +27,8 @@ def __init__(self, tree, *args, **kwargs):
         'C406': 'C406 Unnecessary {type} literal - rewrite as a dict literal.',
         'C407': "C407 Unnecessary list comprehension - '{func}' can take a generator.",
+        'C408': 'C408 Unnecessary {type} literal - rewrite as a tuple literal.',
+        'C409': 'C409 Unnecessary {type} literal - rewrite as a list literal.',
```
I think that these need rewording? Perhaps:

```diff
+        'C408': 'C408 Unnecessary {type} literal passed to tuple constructor - rewrite as a tuple literal.',
+        'C409': 'C409 Unnecessary {type} literal passed to list constructor - rewrite as a list literal.',
```

Maybe a little wordy and perhaps "constructor" isn't the best term, but should be clearer...

adamchainz replied:
"passed to tuple()" ?

jdufresne commented:
Thanks for the review. I have updated the wording as suggested.

adamchainz commented:
Can you rebase, now I've merged #51 this needs the error numbers C409 and C410. Additionally when hitting C410 we'll need the entry point in `setup.py` to be updated from `C40 =` to allow for the new code - I'm not sure if there's an easy way of specifying C40X and C41X are registered by the plugin 🤔 .

jdufresne commented:
Rebased and adjusted the entry point. Thanks.

pope1ni commented on README.rst:
```diff
@@ -106,14 +108,3 @@ The list of builtins that are checked for are:
 * ``sum``
 * ``tuple``
-
-C408: Unnecessary (dict/list/tuple) call - rewrite as a literal.
```
I presume that this removal is unintentional after the rebase? Should a section for C409-C410 be added?

adamchainz replied:
Indeed it should!

jdufresne commented:
Thanks for the review. Good catch. I have updated the commit:

* Restored deleted README section
* Added details to README about the new rules

Cheers.

jdufresne commented:
@adamchainz Do you feel your comment in #63 applies here as well. "rewrite as list literal" may be confusing as there is already a list literal inside the `list()` call.

adamchainz commented:
Good point, of course you're also detecting when there's a list literal inside a call to `tuple`, so to do it clearly you might need some logic around the message. Shouldn't be that bad though 🤔

jdufresne commented:
Updated with suggested changes.
